pkg/utils/handler.go

```diff
@@ -164,14 +257,18 @@ func (h *Handler) CheckAgents() ([]string, []string, error) {
 	}
 	for _, pod := range pods.Items {
 		agentName := pod.ObjectMeta.Name
-		agentData, exists := h.AgentCache[agentName]
-		if !exists {
+		agent, err := h.ExtensionsClientset.Agents().Get(agentName)
```

Comment:
It has a big impact on connectivity check speed (it was about 5 second for 3 nodes on the lab).